File: Developer-Examples/hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/ActionButtonActivity.java
/**
 * RealWear Development Software, Source Code and Object Code
 * (c) RealWear, Inc. All rights reserved.
 * <p>
 * Contact info@realwear.com for further information about the use of this code.
 */

package com.realwear.hmt1developerexamples;

import android.app.Activity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.WindowManager;
import android.widget.ImageView;

/**
 * Activity that shows how to interact with the action button on a HMT-1 device
 */
public class ActionButtonActivity extends Activity {
    private static final int ActionBtnKeyCode = 500;

    private ImageView mActionButtonImageView;

    /**
     * Called when the activity is created
     *
     * @param savedInstanceState See Android docs
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.action_button_main);

        mActionButtonImageView = (ImageView) findViewById(R.id.actionButtonImageView);

        // Disable the action button default behaviour
        mActionButtonImageView.setContentDescription("hf_no_ptt_home");
    }

    /**
     * Called when activity is resumed - See Android docs
     */
    @Override
    protected void onResume() {
        super.onResume();
        mActionButtonImageView.setImageResource(R.drawable.radio_button_unchecked);
    }

    /**
     * Called when activity is paused - See Android docs
     */
    @Override
    protected void onPause() {
        super.onPause();
        this.finish();
    }

    /**
     * Called when a key is pressed - See Android docs
     *
     * @param keyCode The value in event.getKeyCode()
     * @param event   Description of the key event
     * @return true if the action button was pressed, otherwise false
     */
    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        switch (keyCode) {
            case ActionBtnKeyCode:
                mActionButtonImageView.setImageResource(R.drawable.radio_button_red);
                return true;
        }

        return super.onKeyDown(keyCode, event);
    }

    /**
     * Called when a key is released - See Android docs
     *
     * @param keyCode The value in event.getKeyCode()
     * @param event   Description of the key event
     * @return true if the action button was pressed, otherwise false
     */
    @Override
    public boolean onKeyUp(int keyCode, KeyEvent event) {
        switch (keyCode) {
            case ActionBtnKeyCode:
                mActionButtonImageView.setImageResource(R.drawable.radio_button_unchecked);
                return true;
        }

        return super.onKeyUp(keyCode, event);
    }
}

File: hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/ActionButtonActivity.kt
/**
 * RealWear Development Software, Source Code and Object Code
 * (c) RealWear, Inc. All rights reserved.
 *
 *
 * Contact info@realwear.com for further information about the use of this code.
 */
package com.realwear.hmt1developerexamples

import android.app.Activity
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.widget.ImageView
import com.realwear.hmt1developerexamples.databinding.ActionButtonMainBinding

/**
 * Activity that shows how to interact with the action button on a HMT-1 device
 */
class ActionButtonActivity : Activity() {
    private val binding: ActionButtonMainBinding by lazy { ActionButtonMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        // Disable the action button default behaviour
        binding.actionButtonImageView.contentDescription = "hf_no_ptt_home"
    }

    override fun onResume() {
        super.onResume()
        binding.actionButtonImageView.setImageResource(R.drawable.radio_button_unchecked)
    }

    override fun onPause() {
        super.onPause()
        finish()
    }

    /**
     * Sets radio button to checked if the [keyCode] is ACTION_BTN_KEY_CODE and returns true,
     * otherwise returns call to superclass' onKeyDown with [keyCode] and [event]
     */
    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        return when (keyCode) {
            ACTION_BTN_KEY_CODE -> {
                binding.actionButtonImageView.setImageResource(R.drawable.radio_button_red)
                true
            }
            else -> super.onKeyDown(keyCode, event)
        }
    }

    /**
     * Sets the radio button to unchecked if the [keyCode] is ACTION_BTN_KEY_CODE and returns true,
     * otherwise returns call to superclass' onKeyUp with [keyCode] and [event]
     */
    override fun onKeyUp(keyCode: Int, event: KeyEvent): Boolean {
        return when (keyCode) {
            ACTION_BTN_KEY_CODE -> {
                binding.actionButtonImageView.setImageResource(R.drawable.radio_button_unchecked)
                true
            }
            else -> super.onKeyUp(keyCode, event)
        }
    }

    companion object {
        private const val ACTION_BTN_KEY_CODE = 500
    }
}